### ISSUE TITLE
Failed testcases in Temporal Deduction with variable

### ISSUE THREAD

**Issue by l591326240** [opennars/opennars#35]
```
IN: <(&&,<$x --> [boy]>,<$x --> [girl]>) =\> (&&,<$x --> [jumping]>,<$x --> [with-wings]>)>. 
IN: <(&&,<$y --> [flying]>,<$y --> bird>) =\> (&&,<$y --> [boy]>,<$y --> [girl]>)>.

Expected outputs:

OUT: <(&&,<$1 --> [flying]>,<$1 --> bird>) =\>  (&&,<$1 --> [jumping]>,<$1 --> [with-wings]>)>.
OUT: <(&&,<$1 --> [jumping]>,<$1 --> [with-wings]>) =/> (&&,<$1 --> [flying]>,<$1 --> bird>)>.

Actual:

No outputs

--------------------------------------------------------------------------------------------------------------

IN: <(&/,<$x --> [boy]>,<$x --> [girl]>) =\> (&/,<$x --> [flying]>,<$x --> [with-wings]>)>. 
IN: <(&/,<$y --> [with-Beak]>,<$y --> bird>) =\> (&/,<$y --> [boy]>,<$y --> [girl]>)>.

Expected outputs:

OUT: <(&/,<$y --> [with-Beak]>,<$y --> bird>) =\> (&/,<$x --> [flying]>,<$x --> [with-wings]>)>.
OUT: <(&/,<$x --> [flying]>,<$x --> [with-wings]>) =/> (&/,<$y --> [with-Beak]>,<$y --> bird>)>.

Actual:

No outputs

```

**Comment by patham9** [comment]
also this is a good find.

Best regards,
Patrick

**Comment by l591326240** [comment]
I think the following picture will be helpful. This is how the concept network looks like for this problem, I think it should be one single network, but for this testcases, the network is separated to two parts.
![image](https://cloud.githubusercontent.com/assets/11791925/7222870/f8aa582e-e749-11e4-8e34-f9821363ae46.png)

**Comment by patham9** [comment]
this means that also terms like (&&,<$1 --> [flying]>,<$1 --> bird>) should be able to name a concept, then the for the above inference needed sub-linking happens and the inference works. I will fix it today.

**Comment by pei2nars** [comment]
That compound term should be taken as a variable term itself, so shouldn't
refer to a concept. This type of inference should happen in its constant
component, like [flying] or bird.

**Comment by patham9** [comment]
In the second case it should be

```
OUT: <(&/,<$x --> [with-Beak]>,<$x --> bird>) =\> (&/,<$x --> [flying]>,<$x --> [with-wings]>)>.
OUT: <(&/,<$x --> [flying]>,<$x --> [with-wings]>) =/> (&/,<$x --> [with-Beak]>,<$x --> bird>)>.
```

instead of

```
OUT: <(&/,<$y --> [with-Beak]>,<$y --> bird>) =\> (&/,<$x --> [flying]>,<$x --> [with-wings]>)>.
OUT: <(&/,<$x --> [flying]>,<$x --> [with-wings]>) =/> (&/,<$y --> [with-Beak]>,<$y --> bird>)>.
```

Fixed, will be part of the next commit in 1.6.4-devel.

**Comment by patham9** [comment]
Fixed in 2c59ddd2fa58753818948d2c8e8d1397ff487be6
